app/helpers/script_helper.rb

```diff
@@ -6,33 +6,28 @@ def javascript_include_tag_without_preload(...)
     without_preload_links_header { javascript_include_tag(...) }
   end
 
-  def javascript_packs_tag_once(*names, prepend: false)
-    @scripts ||= []
-    if prepend
-      @scripts = names | @scripts
-    else
-      @scripts |= names
-    end
+  def javascript_packs_tag_once(*names, **attributes)
+    @scripts = @scripts.to_h.merge(names.index_with(attributes))
     nil
   end
 
   alias_method :enqueue_component_scripts, :javascript_packs_tag_once
 
-  def render_javascript_pack_once_tags(*names)
-    names = names.presence || @scripts
-    if names && (sources = AssetSources.get_sources(*names)).present?
-      safe_join(
-        [
-          javascript_assets_tag(*names),
-          *sources.map do |source|
-            javascript_include_tag(
-              source,
-              crossorigin: local_crossorigin_sources? ? true : nil,
-              integrity: AssetSources.get_integrity(source),
-            )
-          end,
-        ],
-      )
+  def render_javascript_pack_once_tags(...)
+    capture do
```

Comment:
Is there a reason this switches to `capture` / `concat`?

Reply:
> Is there a reason this switches to `capture` / `concat`?

This was more stylistic / refactoring, not strictly necessary. I either forgot or hadn't been aware of `capture`/`concat`, and it felt a little more ergonomic to additively append to the output buffer, vs. `safe_join` and ensuring that whatever's concatenated is the last statement of the method (the return value).

Reply:
Works for me!